cdap-ui/cypress/integration/nuxtour.spec.ts

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License'); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import * as Helpers from '../helpers';
+
+let headers = {};
+const controlCenterText = 'Control Center';
+
+// Assumes 5 step tour: Control Center, Pipeline Studio, Wrangler, Metadata, Hub
+
+describe('NUX tour tests', () => {
+  before(() => {
+    Helpers.loginIfRequired().then(() => {
+      cy.getCookie('CDAP_Auth_Token').then((cookie) => {
+        if (!cookie) {
+          return;
+        }
+        headers = {
+          Authorization: 'Bearer ' + cookie.value,
+        };
+      });
+      // cy.request({
+      //   method: 'GET',
+      //   url: `http://${Cypress.env('host')}:11015/v3/configuration/user`,
+      // }).then((res) => {
+      //   expect(res.body).to.not.have.property('showWelcome');
+      // });
+    });
+  });
+  after(() => {
+    cy.window().then((win) => {
+      win.sessionStorage.removeItem('nuxTesting');
+    });
+    cy.request({
+      method: 'PUT',
+      url: `http://${Cypress.env('host')}:11015/v3/configuration/user`,
+      body: { property: {} },
+      headers,
+    });
+  });
+  beforeEach(() => {
+    cy.visit('/cdap', {
+      onBeforeLoad: (win) => {
+        win.sessionStorage.clear();
+        win.sessionStorage.setItem('nuxTesting', 'true');
+      },
+    });
+  });
+
+  it('Should show 5 step tour when user first visits homepage', () => {
+    cy.get('[data-cy="welcome-nux-tour"]').should('exist');
+    cy.get('[data-cy="start-tour-btn"]').click();
+
+    cy.get('.shepherd-title').then((modalText) => {
+      expect(modalText).to.contain(controlCenterText);
+    });
+
+    cy.get('[data-id="control-center"]').within(() => {
+      cy.get('.next-btn').click();
+    });
+
+    cy.get('.shepherd-title').then((modal) => {
+      const pipelinesText = 'Pipeline Studio';
+      expect(modal).to.contain(pipelinesText);
+    });
+
+    cy.get('[data-id="pipelines"]').within(() => {
+      cy.get('.next-btn').click();
+    });
+
+    cy.get('.shepherd-title').then((modal) => {
+      const dataPrepText = 'Wrangler';
+      expect(modal).to.contain(dataPrepText);
+    });
+
+    cy.get('[data-id="preparation"]').within(() => {
+      cy.get('.next-btn').click();
+    });
+
+    cy.get('.shepherd-title').then((modal) => {
+      const metadataText = 'Metadata';
+      expect(modal).to.contain(metadataText);
```

Comment:
Since we are just using this variable in the next line it doesn't make much sense to have it in a variable. We can as well use it directly.